### PR TITLE
add flask server and api

### PR DESCRIPTION
**Pull request by yujinjcho**
https://github.com/yujinjcho/movie_recommendations/issues/25

### CODE REVIEW

File: movie_rec/recommender_server.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import json
+from optparse import OptionParser
+import psycopg2
+
+class MovieRecommender(object):
```
Comment (achiu8) [inline movie_rec/recommender_server.py]: Should this class name match the filename? What's the difference between this and the existing `MovieRecommender`?

File: movie_rec/recommender_server.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    def load_mapping(self):
+        conn = psycopg2.connect('dbname=movie_rec')
+        cur = conn.cursor()
+        query = "select movie_id, title from movies"
+        cur.execute(query)
+        results = cur.fetchall()
+        cur.close()
+        conn.close()
+        return dict([
```
Comment (achiu8) [inline movie_rec/recommender_server.py]: `return dict(results)`?

File: movie_rec/recommender_server.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        ])
+
+
+    def top_n(self, n):
+        predict_like = [
+            (movie,scores) for movie, scores in self.predictions.items() 
+            if scores[1] > scores[0]
+        ]
+        predict_sorted = sorted(predict_like, key=lambda x: x[1][1], reverse=True)
+        predict_sorted = [x for x in predict_sorted if x[0] not in self.seen_movies]
```
Comment (achiu8) [inline movie_rec/recommender_server.py]: Didn't we combine this filtering with the above comprehension?

File: movie_rec/recommender_server.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        return self.recommendations(predict_sorted, n)
+
+
+    def bottom_n(self, n):
+        predict_dislike = [
+            (movie,scores) for movie, scores in self.predictions.items() 
+            if scores[0] > scores[1]
+        ]
+        predict_sorted = sorted(predict_dislike, key=lambda x: x[1][0], reverse=True)
```
Comment (achiu8) [inline movie_rec/recommender_server.py]: Filter for not seen?

File: movie_rec/recommender_server.py (hunk 1 above)
Comment (yujinjcho) [reply movie_rec/recommender_server.py]: i don't think there is strict convention in python. just got rid of it for the server one.

**Comment by yujinjcho** [comment]
looks like forgot to merge dev

**Pull request merged**